pycbc/inference/sampler/refine.py

```diff
@@ -0,0 +1,234 @@
+""" Sampler that uses kde refinement of an existing posterior estimate.
+"""
+
+import logging
+import numpy
+import numpy.random
+
+from scipy.special import logsumexp
+from scipy.stats import gaussian_kde
+from scipy.stats import entropy as sentropy
+
+from pycbc.inference import models
+from pycbc.pool import choose_pool
+from pycbc.inference.io import loadfile
+
+from .base import setup_output, initial_dist_from_config
+from .dummy import DummySampler
+
+
+def call_model(params):
+    models._global_instance.update(**params)
+    return (models._global_instance.logposterior,
+            models._global_instance.loglikelihood)
+
+
+def resample_equal(samples, logwt, seed=0):
```

Comment:
Please add a docstring for this function. I think it's important enough that we should know what it does. (sorry these comments come late. I wrote them while github was misbehaving, and it looks like they got swallowed!)